codegen/module.go

```diff
@@ -1024,215 +1022,80 @@ func (system *ModuleSystem) getSpec(instance *ModuleInstance) (interface{}, bool
-// or endpoint configuration) of the endpoint will not vary the output of the service's template. This allows us to make
-// a simplifying assumption about running incremental build: we only need to run codegen for module X if the configuration
-// file for X has changed. If the assumption was false, then we would need to run codegen for module X if any of the
-// dependencies of X have changed, which would require pulling the dependencies of X (i.e. the YAML *-config.yaml files
-// ) out into the build system.
-func (system *ModuleSystem) tryResolveIncrementalBuild(instances []ModuleDependency, resolvedModules map[string][]*ModuleInstance) (map[string][]*ModuleInstance, error) {
-	// toBeBuiltModules is the list of module instances affected by this build
-	toBeBuiltModules := make(map[string]map[string]*ModuleInstance, 0)
-	for _, className := range system.classOrder {
-		toBeBuiltModules[className] = make(map[string]*ModuleInstance, 0)
-	}
-
-	for _, className := range system.classOrder {
-		classInstances := resolvedModules[className]
-
-		for _, classInstance := range classInstances {
-			// Some generators require dependent tasks to have computed their specs.
-			spec, ok, err := system.getSpec(classInstance)
-			if err != nil {
-				// Generators must be adapted to the new SpecProvider interface so that
-				// specs can be computed. Fall back to full build if generator errors when
-				// computing the spec or is not adapted to the SpecProvider interface.
-
-				return nil, err
-			}
-
-			if ok {
-				classInstance.genSpec = spec
-			}
-
-			found := false
-			for _, instance := range instances {
-				if classInstance.InstanceName == instance.InstanceName && classInstance.ClassName == instance.ClassName {
-					found = true
-					break
-				}
-			}
-			if !found {
-				fmt.Printf(
-					"Skipping generation of %q %q class of type %q "+
-						"as not needed for incremental build\n",
-					classInstance.InstanceName,
-					classInstance.ClassName,
-					classInstance.ClassType,
-				)
-				continue
-			}
-
-			toBeBuiltModules[classInstance.ClassName][classInstance.InstanceName] = classInstance
-		}
-	}
-
-	toBeBuiltModulesList := make(map[string][]*ModuleInstance)
-	for _, className := range system.classOrder {
-		toBeBuiltModulesList[className] = make([]*ModuleInstance, 0, len(toBeBuiltModules[className]))
-		for _, classInstance := range toBeBuiltModules[className] {
-			toBeBuiltModulesList[className] = append(toBeBuiltModulesList[className], classInstance)
-		}
-	}
-
-	return toBeBuiltModulesList, nil
-}
-
-// GenerateIncrementalBuild is like GenerateBuild but filtered to only the given
-// module instances.
-func (system *ModuleSystem) GenerateIncrementalBuild(
+// Build invokes the generator for a module instance and optionally writes the files to disk
+func (system *ModuleSystem) Build(
 	packageRoot string,
 	baseDirectory string,
+	physicalGenDir string,
 	targetGenDir string,
```

Comment:
Is this still necessary with the newly added `physicalGenDir`?

Reply:
You're right it can be removed here, we used to call `ResolveModules` but now we don't.